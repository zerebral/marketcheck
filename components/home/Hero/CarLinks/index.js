import styled from 'styled-components'
import { rem } from 'polished'
import PropTypes from 'prop-types'
import CarLinkAnchor from '@/Home/categories/CarLinkAnchor'
import { mediaMax } from '%/styles/mixins'

const CarLinks = styled.ul`
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  list-style: none;
  ${mediaMax.laptop`
    flex-wrap: wrap;
  `}
`

const CarLink = styled.li`
  font-size: ${rem(14)};

  & + & {
    margin-left: ${rem(20)};
    ${mediaMax.tablet`
      margin-left: 0;
    `}
  }

  ${mediaMax.tablet`
    margin-bottom: ${rem(10)};
    padding: 0 ${rem(5)};
  `}
`

const Rendered = ({links}) => (
  <CarLinks>
    {links.map((link, index) => (
      <CarLink key={link + index}>
          <CarLinkAnchor link={link}/>
      </CarLink>
    ))}
  </CarLinks>
)

Rendered.propTypes = {
  links: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    route: PropTypes.string.isRequired
  })).isRequired
}

export default Rendered
