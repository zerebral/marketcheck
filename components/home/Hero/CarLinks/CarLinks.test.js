/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import CarLinks from './index'

describe('CarLinks Component', () => {
  test('CarLinks Passes shallow snapshot', () => {
    const links = [
      { name: 'SUV', route: '#' },
      { name: 'Sedan', route: '#' },
      { name: 'Pickup', route: '#' },
      { name: 'Coupe', route: '#' },
      { name: 'Hatchback', route: '#' },
      { name: 'Convertible', route: '#' },
      { name: 'Wagon', route: '#' },
      { name: 'Crossover', route: '#' }
    ]
    const wrapper = shallow(<CarLinks links={links} />)
    expect(wrapper).toMatchSnapshot()
  })
})
