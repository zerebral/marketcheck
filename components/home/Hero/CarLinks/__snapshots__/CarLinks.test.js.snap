// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarLinks Component CarLinks Passes shallow snapshot 1`] = `
<CarLinks>
  <CarLinks__CarLink>
    <a
      href="#"
    >
      SUV
    </a>
  </CarLinks__CarLink>
  <CarLinks__CarLink>
    <a
      href="#"
    >
      Sedan
    </a>
  </CarLinks__CarLink>
  <CarLinks__CarLink>
    <a
      href="#"
    >
      Pickup
    </a>
  </CarLinks__CarLink>
  <CarLinks__CarLink>
    <a
      href="#"
    >
      Coupe
    </a>
  </CarLinks__CarLink>
  <CarLinks__CarLink>
    <a
      href="#"
    >
      Hatchback
    </a>
  </CarLinks__CarLink>
  <CarLinks__CarLink>
    <a
      href="#"
    >
      Convertible
    </a>
  </CarLinks__CarLink>
  <CarLinks__CarLink>
    <a
      href="#"
    >
      Wagon
    </a>
  </CarLinks__CarLink>
  <CarLinks__CarLink>
    <a
      href="#"
    >
      Crossover
    </a>
  </CarLinks__CarLink>
</CarLinks>
`;
